### ISSUE TITLE
Failed to connect: HTTP response 500

### ISSUE THREAD

**Issue by JunyuanZhangCoin** [databricks/databricks-sql-go#23]
The issue was when I tried to query Databricks using the Go sql driver (https://docs.databricks.com/dev-tools/go-sql-driver.html)
I got the following error: driver: bad connection: HTTP Response code: 500 . It was flaky at first but now it is 100% failing. After further debugging, I found that the error happened when trying to acquire a connection, before actually sending the query.

the error is thrown from here
https://github.com/databricks/databricks-sql-go/blob/main/hive/client.go#L43

error : error(*[github.com/apache/thrift/lib/go/thrift.tTransportException](https://github.com/apache/thrift/lib/go/thrift.tTransportException)) *{typeId: 0, err: error(*errors.errorString) *{s: "HTTP Response code: 500"}}

**Comment by ericsun2** [comment]
A more meaningful Thrift error code is needed with server side retry mechanism.

**Comment by jadewang-db** [comment]
@JunyuanZhangCoin  what is your workspace name? is this happens on all your workspace or just for this workspace?

**Comment by JunyuanZhangCoin** [comment]
It happens in the dev workspace: coinbase-development.cloud.databricks.com. Didn't try it in the production workspace cause the code hasn't been merged into master.

**Comment by jadewang-db** [comment]
@JunyuanZhangCoin  do you have a timeframe when this error happens?

**Comment by moderakh** [comment]
could it be that we are not closing sessions/resources and we ran out of resources? 
@susodapop are the symptoms similar to pysql bug you worked on (https://github.com/databricks/databricks-sql-python/pull/38)?

**Comment by JunyuanZhangCoin** [comment]
The errors showed up since last Tuesday (9/13) and kept happening

**Comment by JunyuanZhangCoin** [comment]
The snippet reproducing the error:
```
	db, _ := sql.Open("databricks", "databricks://:dapi***************@coinbase-development.cloud.databricks.com/sql/1.0/endpoints/d72dcf4b8344ab11")
	conn, _ := db.Conn(context.Background())
	_, err := conn.QueryContext(context.Background(), "select * from fqs_test.user_features LIMIT 1")
	assert.NotNil(t, err)
```

**Comment by mohit-zs** [comment]
Hey, I am also facing the same issue (bad connection) while trying to connect on my dev environment.

the error I am getting is
Error - driver: bad connection: Post \"https://token:***@[Workspace_hostname]:443/[Endpoint HTTP Path]

otherwise, it is able to connect to the same host via my local machine

**Comment by andrefurlan-db** [comment]
Hi @mohit-zs , @JunyuanZhangCoin. Is this still an issue? Can you try with the latest version and let me know? Check the DNS changes to make sure it works or use the With.. option connector functions

**Comment by mohit-zs** [comment]
@andrefurlan-db I am no longer facing the issue. It was a firewall issue for me.

**Comment by andrefurlan-db** [comment]
Okay, I'll close this issue for now. If anyone still has any problems, please open a new issue. Thanks!

**Comment by rahulanand90** [comment]
I am still facing the same issue. 
`databricks://:dapi*****************************a@dbc-*******-******.cloud.databricks.com/sql/1.0/endpoints/************`
 and then I am passing the DSN to the below function.
`sql.Open("databricks", dsn)`

One observation, things were working fine when I was using 
`github.com/databricks/databricks-sql-go v0.1.4
github.com/apache/thrift v0.12.0 // indirect
`
thrift version depends on databricks-sql-go, so definitely something with wrong withe the latest version of databricks-sql-go


ERROR:
`Error: invalid DSN: invalid DSN port: strconv.Atoi: parsing "": invalid syntax`

**Comment by rahulanand90** [comment]
Thanks @tauseef-h but I have already tried this. I created a new .env file with 
`DATABRICKS_PORT=443`

and then also checked that it is giving me the same in the print statement.
```
_ := godotenv.Load()
databricksPort := os.Getenv("DATABRICKS_PORT")
fmt.Println(databricksPort)
```

I would like to stress again this is giving me errors only after the upgrade as mentioned above.

**Comment by rahulanand90** [comment]
there is a change in dsn format and that was the problem. 

earlier this was working with
```
databricks://:dapi*****************************a@dbc-*******-******.cloud.databricks.com/sql/1.0/endpoints/************
```
but now it should be 
```
token:dapi*****************************a@dbc-*******-******.cloud.databricks.com/sql/1.0/endpoints/************

```
